### DISCUSSION TITLE
Concurrency with Environment Manual Approvals

### DISCUSSION THREAD

**Discussion by leviwolfe** [community/community#17401] [Actions]
This seems like an unexpected behavior for jobs that uses both concurrency and an environment that requires manual approval:

1. The first workflow run will properly require approval. Leave it waiting.
2. A second workflow run will be blocked and cannot be approved until the first workflow run is approved.

In this situation, I would not expect concurrency to take effect until after the approval. If the first workflow is approved, then the 2nd run could be pending. The second workflow should be able to be approved, and if so, then it should be able to cancel the first.

The cancel-in-progress flag does not really solve this problem, because there are situations where I want to be able to approve the first run, and situations where I want to skip the first run and directly approve the second run.

**Comment by leviwolfe** [comment]
I had found [this similar comment](https://github.community/t/bug-with-concurrency-environment-approve/180985) but no follow-up here.

**Comment by devminded** [comment]
Unfortunately this is how it works.

If I understood your scenario correct, there is a workaround:

1. On the job with Environment approvals - set **no concurrency** and just a dummy step with "echo approved".
2. Have a downstream job that does the actual work, with "needs" on the approval workflow.
3. You set your concurrency policy on the downstream job.
